### PR TITLE
store/tikv:remove "tidb/tablecodec"  from snapshot.go

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR removes package "tidb/tablecodec" from "store/tikv/snapshot.go".
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/txn.go [hunk 1]
```diff
@@ -471,7 +470,12 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tidbkv.LockCtx, keysInp
 					txn.us.UnmarkPresumeKeyNotExists(key)
 				}
 			}
-			keyMayBeLocked := terror.ErrorNotEqual(tidbkv.ErrWriteConflict, err) && terror.ErrorNotEqual(tidbkv.ErrKeyExists, err)
+			_, ok := errors.Trace(err).(*kv.ErrWriteConflict)
```
Comment (sticnarf) [inline store/tikv/txn.go:473]: `errors.Cause`?

File: store/driver/txn/error.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		return nil
+	}
+	if e, ok := errors.Cause(err).(*tikvstore.ErrWriteConflict); ok {
+		return newWriteConflictError(e.WriteConflict)
+	}
+	if e, ok := errors.Cause(err).(*tikvstore.ErrRetryable); ok {
+		notFoundDetail := prettyLockNotFoundKey(e.Retryable)
+		return kv.ErrTxnRetryable.GenWithStackByArgs(e.Retryable + " " + notFoundDetail)
+	}
+	return errors.Trace(err)
```
Comment (sticnarf) [inline store/driver/txn/error.go:148]: As there could be more and more types of errors, what about using `switch errors.Cause(err).(type)`?

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by disksing — APPROVED** [review]

**Pull request merged**